### ISSUE TITLE
Fix false positives interpolation in function-calc-no-invalid

### ISSUE THREAD

**Issue by AlexWayfer** [stylelint/stylelint#4336]
> Clearly describe the bug

There are false positives for two complex SCSS/SASS expressions in `calc()`.

> Which rule, if any, is the bug related to?

`function-calc-no-invalid`

> What CSS is needed to reproduce the bug?

```scss
left: calc(
  -#{$li_padding} +
  #{$button_padding}#{if($button_padding == 0, 'px', '')} -
  #{$border_width}
);
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```yaml
rules:
  function-calc-no-invalid: true
```

> Which version of stylelint are you using?

`11.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint test.scss`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS variables, interpolation and ternary `if`.

> What did you expect to happen?

No errors to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following errors were flagged:

```shell
test.scss
 2:4  ✖  Expected a valid expression   function-calc-no-invalid
```

**Comment by evilebottnawi** [comment]
Thanks for issue,

@stylelint/core i think we should rewrite rule, ignore any `calc` what contains non standard syntax, also latest `postcss-value-parser` parse calc according spec so we can drop parser

**Comment by antokara** [comment]
I got the same false positive with styled components:

```js
const MyStyledComponent = styled.div`
  margin: calc(-${props => props.margin} / 5);
`;
```

where `props.margin` equals `10em`.

AFAIK, this should be a valid calc...
It used to work in version `10.1.0` without issues.

**Comment by in-in** [comment]
I'm facing this problem, too. :disappointed: 
The minus sign and  SCSS variable interpolation cause an error.

```css
a {
  width: calc((-#{$breakpoint} - -100%) * 999);
}
```

```json
{
  "syntax": "scss",
  "rules": {
    "function-calc-no-invalid": true
  }
}
```

Demo: [https://stylelint.io/demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYAdAdilB3AllAC4AW8KYSANmABS0C0AxMACQBGATjEgNYAOEfJiIBfFAwkBGAAwyApAEoUAKhQBOTYoDcWUSAA0IAM4BPEUgAeCcMeOHwETADN8AcxsZsKdCfNErX3JfYzA7XwMsHF9OAFcqGGMgtCicHxBnWMwwInwnBkoaBkwIBmEAN2pCZKI4mFTRPRBRIA)

**Comment by daton89** [comment]
I'm having the same problem with `styled-components`: 

```js
css`
      max-width: calc(100% - ${isLarge ? xl : lg});
`}
```

**Comment by jeddy3** [comment]
> @evilebottnawi i think we should rewrite rule, ignore any calc what contains non standard syntax, also latest postcss-value-parser parse calc according spec so we can drop parser

SGTM. Labeling as "help wanted". If anyone has time, feel free to contribute this reworking via a PR and it will address the bug.

**Comment by jeddy3** [comment]
Closing as duplicate of https://github.com/stylelint/stylelint/issues/4114
